src/main/resources/templates/db_scripts/transactions.sql

```diff
@@ -1,44 +1,47 @@
 -- https://www.postgresql.org/docs/10/functions-json.html
 CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.recalculate_totals() RETURNS TRIGGER AS $recalculate_totals$
   DECLARE
-    fromBudget              jsonb;
-    fromBudgetAllocated     decimal;
-    fromBudgetAvailable     decimal;
-    fromBudgetUnavailable   decimal;
-
-    fromLedger              jsonb;
-    fromLedgerAllocated     decimal;
-    fromLedgerAvailable     decimal;
-    fromLedgerUnavailable   decimal;
-
-    toBudget                jsonb;
-    toBudgetAllocated       decimal;
-    toBudgetAvailable       decimal;
-    toBudgetUnavailable     decimal;
-
-    toLedger                jsonb;
-    toLedgerAllocated       decimal;
-    toLedgerAvailable       decimal;
-    toLedgerUnavailable     decimal;
-
-    newBudgetValues         text[];
-    newLedgerValues         text[];
-    amount                  decimal;
-    transactionType         text;
+    fromBudget                  jsonb;
+    fromBudgetAllocated         decimal;
+    fromBudgetAvailable         decimal;
+    fromBudgetUnavailable       decimal;
+    fromBudgetEncumbered        decimal;
+    fromBudgetOverEncumbered    decimal;
+
+    fromLedger                  jsonb;
+    fromLedgerAllocated         decimal;
+    fromLedgerAvailable         decimal;
+    fromLedgerUnavailable       decimal;
+
+    toBudget                    jsonb;
+    toBudgetAllocated           decimal;
+    toBudgetAvailable           decimal;
+    toBudgetUnavailable         decimal;
+
+    toLedger                    jsonb;
+    toLedgerAllocated           decimal;
+    toLedgerAvailable           decimal;
+    toLedgerUnavailable         decimal;
+
+    newBudgetValues             text[];
+    newLedgerValues             text[];
+    amount                      decimal;
+    transactionType             text;
 
     BEGIN
       amount = (NEW.jsonb->>'amount')::decimal;
       transactionType = NEW.jsonb->>'transactionType';
 
-      -- Recalculate Allocation / Transfer
-      IF (transactionType = 'Allocation' OR transactionType = 'Transfer') THEN
+      -- Recalculate Allocation / Transfer / Encumbrance
+      IF (transactionType = 'Allocation' OR transactionType = 'Transfer' OR transactionType = 'Encumbrance') THEN
         --check if fromFundId exists
         IF (NEW.jsonb->'fromFundId' IS NOT NULL) THEN
```

Comment:
I think it makes sense to raise an error if `fromFundId` is not specified for the 
Encumbrance.